subsys/usb/class/dfu/usb_dfu.c

```diff
@@ -461,11 +497,10 @@ static int dfu_class_handle_req(struct usb_setup_packet *pSetup,
 			dfu_reset_counters();
 			k_poll_signal_reset(&dfu_signal);
 
-			if (dfu_data.flash_area_id !=
-			    UPLOAD_FLASH_AREA_ID) {
+			if (dfu_partition_readonly(dfu_data.fa->fa_id)) {
 				dfu_data.status = errWRITE;
 				dfu_data.state = dfuERROR;
-				LOG_ERR("This area can not be overwritten");
+				LOG_ERR("DFU_DNLOAD area is readonly");
```

Comment:
typo DFU_DNLOAD -> DFU_DOWNLOAD

Reply:
All the other print messages use `DFU_DNLOAD`, should they all be `DF_DOWNLOAD`?

Reply:
DFU_DNLOAD is a DFU class request. Maybe:
```suggestion
				LOG_ERR("DFU_DNLOAD write area is readonly");
```